community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/index/DeferredConstraintVerificationUniqueLuceneIndexPopulator.java

```diff
@@ -139,17 +138,18 @@ public IndexUpdater newPopulatingUpdater( final PropertyAccessor accessor ) thro
             List<Object> updatedPropertyValues = new ArrayList<>();
 
             @Override
-            public void process( NodePropertyUpdate update ) throws IOException, IndexEntryConflictException
+            public PreparedIndexUpdates prepare( Iterable<NodePropertyUpdate> updates ) throws IOException
```

Comment:
This seems to apply the updates in the prepare phase, before we know if the transaction is to be committed or not. That's Bad™. Rather I'd think that this method should return a `PreparedIndexUpdates` that would perform that task when `commit()` is invoked.